pkgs/development/python-modules/open-clip-torch/default.nix

```
@@ -0,0 +1,55 @@
+{ lib
+, buildPythonPackage
+, ftfy
+, huggingface-hub
+, protobuf
+, regex
+, sentencepiece
+, timm
+, torch
+, torchvision
+, tqdm
+, setuptools
+, wheel
+, fetchFromGitHub
+,
+}:
+buildPythonPackage rec {
+  pname = "open-clip-torch";
+  version = "2.23.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mlfoundations";
+    repo = "open_clip";
+    rev = "v${version}";
+    hash = "sha256-Txm47Tc4KMbz1i2mROT+IYbgS1Y0yHK80xY0YldgBFQ=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    ftfy
+    huggingface-hub
+    protobuf
+    regex
+    sentencepiece
+    timm
+    torch
+    torchvision
+    tqdm
+  ];
+
+  pythonImportsCheck = [ "open_clip" ];
+
+  meta = with lib; {
+    description = "An open source implementation of CLIP";
+    homepage = "https://github.com/mlfoundations/open_clip";
+    license = licenses.mit;
```

```suggestion
    license = licenses.asl20;
```
https://github.com/mlfoundations/open_clip/blob/29b90b8dde9125c479c0ddbab477310f665a0435/setup.py#L39